evalml/tests/component_tests/decomposer_tests/test_polynomial_decomposer.py

```diff
@@ -398,19 +399,59 @@ def test_polynomial_decomposer_uses_time_index(
             assert isinstance(y.index, pd.DatetimeIndex)
 
 
+def generate_seasonal_data(
+    period,
+    step=None,
+    num_periods=10,
+    scale=1,
+    trend_degree=1,
+    real_or_synthetic="synthetic",
+):
+    """Function that returns data with a linear trend and a seasonal signal with specified period."""
+
+    def generate_synthetic_data(period, step, num_periods, scale, trend_degree):
+        if period is None:
+            x = np.arange(0, 1, 0.01)
+        elif step is not None:
+            freq = 2 * np.pi / period / step
+            x = np.arange(0, 1, step)
+        else:
+            freq = 2 * np.pi / period
+            x = np.arange(0, period * num_periods, 1)
+        dts = pd.date_range(datetime.today(), periods=len(x))
+        X = pd.DataFrame({"x": x})
+        X = X.set_index(dts)
+
+        if trend_degree == 1:
+            y_trend = pd.Series(scale * minmax_scale(x + 2))
+        elif trend_degree == 2:
+            y_trend = pd.Series(scale * minmax_scale(x**2))
+        elif trend_degree == 3:
+            y_trend = pd.Series(scale * minmax_scale((x - 5) ** 3 + x**2))
+        if period is not None:
+            y_seasonal = pd.Series(np.sin(freq * x))
+        else:
+            y_seasonal = pd.Series(np.zeros(len(x)))
+        y = y_trend + y_seasonal
+        return X, y
+
+    if real_or_synthetic == "synthetic":
+        X, y = generate_synthetic_data(period, step, num_periods, scale, trend_degree)
+    elif real_or_synthetic == "real":
+        X, y = load_weather()
+        y = y.set_axis(X["Date"]).asfreq(pd.infer_freq(X["Date"]))
+        X = X.set_index("Date").asfreq(pd.infer_freq(X["Date"]))
+    return X, y
```

Comment:
Without looking at this closely, is there any value in initializing woodwork here?